lib/vquic/ngtcp2.c

```diff
@@ -714,41 +714,119 @@ static int cb_h3_stream_close(nghttp3_conn *conn, int64_t stream_id,
 
   stream->closed = TRUE;
   Curl_expire(data, 0, EXPIRE_QUIC);
+  /* make sure that ngh3_stream_recv is called again to complete the transfer
```

Comment:
This was causing the transfer to never complete if the server didn't send any packets after the final batch that completed the response body. That's because `Curl_readwrite` would only call us back if the socket was readable, but in that case it would never become readable again. Sometimes we would get lucky and the server would send back an ACK of some kind, making the socket readable and letting us complete the transfer, but other times we would not.